src/archivematicaCommon/lib/env_configparser.py

```diff
@@ -0,0 +1,73 @@
+import os
+import ConfigParser
+import functools
+
+
+def fallback_option(fn):
+    def wrapper(*args, **kwargs):
+        fallback = kwargs.pop('fallback', None)
+        try:
+            return fn(*args, **kwargs)
+        except (ConfigParser.NoSectionError, ConfigParser.NoOptionError):
+            if fallback:
+                return fallback
+            raise
+    return functools.wraps(fn)(wrapper)
+
+
+class EnvConfigParser(ConfigParser.SafeConfigParser):
+    """
+    EnvConfigParser enables the user to provide configuration defaults using
+    the string environment, e.g. given:
+
+      - String environment prefix (prefix) = "ARCHIVEMATICA_MCPSERVER"
+      - Configuration section: "network"
+      - Configuration option: "tls"
+
+    This parser will first try to find the configuration value in the string
+    environment matching one of the two following keys:
+
+      - ARCHIVEMATICA_MCPSERVER_NETWORK_TLS
+      - ARCHIVEMATICA_MCPSERVER_TLS
+
+    It the variable is not set in the string environment the reader falls back
+    on the main configuration backend.
+
+    Additionally, the getters (get(), getint(), etc...) accept a new parameter
+    (fallback) that returns the value given to it instead of an exception when
+    the section or option trying to be match are undefined.
+    """
+    ENVVAR_SEPARATOR = '_'
+
+    def __init__(self, defaults=None, env=None, prefix=''):
+        self._environ = os.environ if env is None else env
+        self._prefix = prefix.rstrip('_')
+        ConfigParser.SafeConfigParser.__init__(self, defaults)
```

Comment:
Preferential to use `super` instead of naming the parent class directly when inheriting.